util/url-encoding/README.md

```diff
@@ -0,0 +1,107 @@
+  fog hostname, then the url's hostname is fog.mobilecoin.org. The amount is
+  human readable as a query parameter. The memo is human readable as a query parameter.
+  This means that if the url appears e.g. in an email, I have some visibility into
+  its contents beyond just b58-encoded bytes. If e.g. the `fog-authority-sig` is missing,
+  I can see that that query parameter isn't there, which may make it easier to debug
+  why a transfer is failing, than if everything were just a b58 blob.
+- Be easy to construct using standard url and base64 javascript libraries.
+  If a third-party web developer has e.g. the protobuf corresponding to someone's
+  public address, following schema in external.proto, they should ideally be able
+  to make a well-formed url-encoded payment request without porting libmobilecoin
+  rust code to their platform.
+- Be easily extensible. If we want to add more fields, we can simply add query parameters.
+
+This is an alternative to the b58-payloads design, with different tradeoffs.
+In experiments, it appears that this format is always 10-20% smaller. This is because
+the b58-payloads is applying a b64-like encoding, meant for binary data, to data like URLs,
+which are not entropy. This causes unnecessary bloat of the encoding, and also makes it less useful
+to humans.
+
+Specification
+=============
+
+The mob URL specified here is formed by starting with the *fog-url*, if any,
+as a base. The fog-url type appears in `mc-util-uri`. (fog-url must be valid to
+use with grpc, so it does not contain any path or query parameters).
+Then, additional items from the public address are added to this url.
+
+Scheme
+------
+
+The url *scheme* is `mob` or `insecure-mob`.
+
+- For a fog-url with scheme "fog", the mob-url scheme is "mob".
+- For a fog-url with scheme "insecure-fog", the mob-url scheme is "insecure-mob".
+- If the fog-url is missing, the mob-url scheme is "mob".
+
+Hostname
+--------
+
+- The *hostname* of the mob-url is the hostname of the fog-url.
+- If the fog-url is missing, the hostname is empty. So in this case, the mob-url begins `mob:///`.
+
+Path
+----
+
+The *path* in the mob-url is `'/'` followed by the base64 encoding of the 64-byte sequence formed by concatenating
+the 32-byte spend-public-key and the 32-byte view-public key.
+
+The url-safe base-64 encoding is used, per RFC 3548, with `-`, `=`, and `_` characters.
+https://tools.ietf.org/html/rfc3548#section-4
+
+Query-parameters
+----------------
+
+The `fog-authority-sig` is encoded as a query parameter using the key `'s'`, and the value is
+the base-64 encoding of the bytes of the signature. The url-safe base-64 encoding is used.
+
+The `fog-report-key`, if any, is encoded as a query parameter using the key `'k'`, and this value is a string.
+This string is encoded as usual per `application/x-www-form-urlencoded`. (In practice it is expected
+that this will be omitted, or will only use characters that don't have to be escaped.)
```

Comment:
I believe one of these should be the anchor/hash/hashtag part of the URL. I have suggested the signature in the past, but I'm currently leaning towards the report key, since it's going to be a byte string that is matched against the contents of the report located at the report server, whereas the signature algorithm could change if we decide to change something else.

Reply:
done -- the report-id is the # now